### PR TITLE
Update Fileprocessing sample to use Session API

### PR DESCRIPTION
**Pull request by yycptt**

### CODE REVIEW

File: cmd/samples/fileprocessing/workflow.go [hunk 1]
```diff
@@ -66,26 +66,34 @@ func SampleFileProcessingWorkflow(ctx workflow.Context, fileID string) (err erro
...
 
-	// The following activities needs to be run on the same host as the first activity.
-	// This is achieved by scheduling them through this host specific task-list.
-	hCtx := workflow.WithTaskList(ctx, fInfo.HostID)
 	// The short schedule to start timeout is needed to ensure that if host is down an activity doesn't
 	// get stuck in a host specific task list.
 	// Note that this timeout is so short to make sample more vivid. In production applications it
 	// is usually higher.
-	hCtx = workflow.WithScheduleToStartTimeout(hCtx, time.Second*2)
+	sessionCtx = workflow.WithScheduleToStartTimeout(sessionCtx, time.Second*2)
```
Comment (mfateev) [inline cmd/samples/fileprocessing/workflow.go]: Is it indeed needed  when using session? I though it is already taken care of.

Comment (yycptt) [reply cmd/samples/fileprocessing/workflow.go]: It's possible that an activity is scheduled before the session is regarded as failed (before the session creation activity heartbeat timeout).

Comment (mfateev) [reply cmd/samples/fileprocessing/workflow.go]: Then we need to cancel activity explicitly upon session failure.

Comment (yycptt) [reply cmd/samples/fileprocessing/workflow.go]: Sure, then each session needs to maintain a list of outstanding activities. I can add that feature.

Comment (yycptt) [reply cmd/samples/fileprocessing/workflow.go]: Oh, actually I don't need to do that. I can simply cancel the sessionCtx, then all the activities using that context will be cancelled.

Comment (mfateev) [reply cmd/samples/fileprocessing/workflow.go]: Cool this makes a lot of sense. Just make sure to document that behavior of the context.

File: cmd/samples/fileprocessing/workflow.go [hunk 2]
```diff
@@ -30,13 +30,12 @@ func init() {
...
 //SampleFileProcessingWorkflow workflow decider
 func SampleFileProcessingWorkflow(ctx workflow.Context, fileID string) (err error) {
-	// step 1: download resource file
 	// step 1: download resource file
 	ao := workflow.ActivityOptions{
 		ScheduleToStartTimeout: time.Second * 5,
 		StartToCloseTimeout:    time.Minute,
 		HeartbeatTimeout:       time.Second * 2, // such a short timeout to make sample fail over very fast
-		RetryPolicy:            &cadence.RetryPolicy{
+		RetryPolicy: &cadence.RetryPolicy{
```
Comment (mfateev) [inline cmd/samples/fileprocessing/workflow.go]: did you run go fmt over this code?

File: cmd/samples/fileprocessing/workflow_test.go [hunk 3]
```diff
@@ -31,6 +32,9 @@ func (s *UnitTestSuite) Test_SampleFileProcessingWorkflow() {
 	env := s.NewTestWorkflowEnvironment()
 	env.SetOnActivityStartedListener(func(activityInfo *activity.Info, ctx context.Context, args encoded.Values) {
```
Comment (mfateev) [inline cmd/samples/fileprocessing/workflow_test.go]: Can we change this test to use activity mocks instead?

File: cmd/samples/fileprocessing/workflow.go (hunk 2 above)
Comment (yycptt) [reply cmd/samples/fileprocessing/workflow.go]: I just checked. This is the output of go fmt. Not sure why the original code has a different format.

File: cmd/samples/fileprocessing/workflow_test.go (hunk 3 above)
Comment (yycptt) [reply cmd/samples/fileprocessing/workflow_test.go]: Inside these activities, there're checks for ensuring the activity is running on the same host as other activities. I am not sure how to use activity mocks to do such checks.

Comment (mfateev) [reply cmd/samples/fileprocessing/workflow_test.go]: Makes sense.

**Pull request merged**